packages/create-schemas/tests/e2e.test.ts

```diff
@@ -0,0 +1,46 @@
+import { readFile } from "node:fs/promises";
+import {
+    createTemporaryFolder,
+    dataFolder,
+    runCompiledBin,
+} from "./fixtures.ts";
+import { describe, test } from "vitest";
+import { join } from "node:path";
+
+const timeout = 30 * 1000; // 30 seconds
+
+describe.concurrent("e2e", () => {
+    test(
+        "officevibe.yaml",
+        async ({ expect, onTestFinished }) => {
+            const tempFolder = await createTemporaryFolder({ onTestFinished });
+
+            const source = join(dataFolder, "officevibe.yaml");
+            const output = join(tempFolder, "output.ts");
+
+            await runCompiledBin({ source, output });
+
+            const result = await readFile(output, "utf-8");
+
+            expect(result).toMatchSnapshot();
+        },
+        timeout
+    );
+
+    test(
+        "petstore.json",
+        async ({ expect, onTestFinished }) => {
+            const tempFolder = await createTemporaryFolder({ onTestFinished });
+
+            const source = join(dataFolder, "petstore.json");
+            const output = join(tempFolder, "output.ts");
+
+            await runCompiledBin({ source, output });
+
+            const result = await readFile(output, "utf-8");
+
+            expect(result).toMatchSnapshot();
+        },
+        timeout
+    );
+});
```

Comment:
If I understand the test correctly, you define the expected output of the type generation in e2e.test.ts.snap and then have the test generate the corresponding schemas. We then crosscheck the generated with the expected outputs?

Reply:
Exactly.

It's pretty barebones, but at least it helps us detect unwanted changes.